code/game/machinery/recharger.dm

```diff
@@ -69,7 +69,9 @@
 		if(istype(I, /obj/item/weapon/gun/energy/gun/nuclear) || istype(I, /obj/item/weapon/gun/energy/crossbow))
 			to_chat(user, SPAN_NOTICE("Your gun's recharge port was removed to make room for a miniaturized reactor."))
 			return
-
+		if(istype(I, /obj/item/weapon/gun/energy/plasma/martyr))
```

Comment:
You shouldn't do it here for this item only. It's bad practice in general to be this specific about particular objects in all purpose machines.